test/python/http_recorder.py

```diff
@@ -0,0 +1,280 @@
+            string_to_find (str): request path
+            cassette_dict (dict): a VCR cassette dictionary
+
+        Returns:
+            Request: VCR's representation of a request.
+        """
+        request_indices = [i for i, request in enumerate(cassette_dict['requests']) if
+                           string_to_find in request.path]
+        return [cassette_dict['responses'][i] for i in request_indices]
+
+    @staticmethod
+    def get_new_id(field, path, id_tracker, type_=str):
+        """
+        Creates a new dummy id (or value) for replacing an existing id (or value).
+
+        Args:
+            field (str): field name is used, in same cases, to create a dummy value.
+            path (str): path of the request is used, in same cases, to create a dummy value.
+            id_tracker (dict): a map of already assigned ids and generated ids.
+            type_ (type): type of the value.
+
+        Returns:
+            str: that is used to replace a value.
+        """
+
+        if type_ == float:
+            return 0.42
+        if type_ == int:
+            return 42
+        dummy_name = 'dummy%s%s' % (path.replace('/', ''), field)
+        count = len(list(filter(lambda x: str(x).startswith(dummy_name), id_tracker.values())))
+        return "%s%02d" % (dummy_name, count + 1)
+
+    @staticmethod
+    def get_maching_dicts(data_dict, map_list):
+        """
+        Find subdicts that are described in map_list.
+        Args:
+            data_dict (dict): in which the map_list is going to be searched.
+            map_list (list): the list of nested keys to find in the data_dict
+
+        Returns:
+            list: a list of dictionaries, each of them matches map_list.
+        """
+        ret = []
+        if not map_list:
+            return ret
+        if isinstance(data_dict, list):
+            _ = [ret.extend(IdRemoverPersister.get_maching_dicts(i, map_list)) for i in data_dict]
+        if isinstance(data_dict, dict):
+            if map_list[0] in data_dict.keys():
+                if len(map_list) == 1:
+                    return [data_dict]
+                else:
+                    ret.extend(
+                        IdRemoverPersister.get_maching_dicts(data_dict[map_list[0]], map_list[1:]))
+        return ret
+
+    @staticmethod
+    def remove_id_in_a_json(jsonobj, field, path, id_tracker):
+        """
+        Replaces in jsonobj (in-place) the field with dummy value (which is constructed with
+        id_tracker, if it was already reaplced, or path, if it needs to be created).
+
+        Args:
+            jsonobj (dict): json dictionary from the response body
+            field (str): string with the field in the response to by replaced
+            path (str): request path
+            id_tracker (dict): a dictionary of the ids already assigned.
+        """
+
+        map_list = field.split('.')
+        for maching_dict in IdRemoverPersister.get_maching_dicts(jsonobj, map_list):
```

Comment:
Typo: it is `matching_dict` and `get_matching_dicts` (notice the `t` in `matching`).